### ISSUE TITLE
Question about the loss function using in your code

### ISSUE THREAD

**Issue by TitleZ99** [CrossmodalGroup/HREM#10]
First of all, I would like to express my respect and gratitude to you for accomplishing such an inspiring work. In the paper, you used a basic triplet loss in the design of the loss function, but used a loss function that includes distanceweightedminer of metric learning in the code. I don't know much about this loss function, is this a loss function you designed or from other work? I would very much like to learn more about this loss function. But I didn't find much information about it. Can you teach me why this loss function is used and provide relevant papers to help me learn? Thanks and salutations again, looking forward to your reply.
![1729003600816](https://github.com/user-attachments/assets/e1974b7e-9a29-482a-95c5-c4902aba2663)

**Comment by darkpromise98** [comment]
The `distanceweightedminer` comes from the previous metric learning work [1].  It will sample the negative samples based on the similarity/distance of **negative pairs**. The greater the similarity, the higher the probability of being selected. You can understand this as a mining strategy of negative samples. 

Compared with the original triplet loss (just select the hardest negative samples), the `distanceweightedminer` selects soft negative samples and bring robust training.

[1] sampling matters in deep embedding learning, ICCV, 2017

**Comment by darkpromise98** [comment]
Besides, the code implementation comes from the official repo [2] (the image/single-modal retrieval task), and  I did minor modification for cross-modal task.

[2] https://github.com/chaoyuaw/incubator-mxnet/blob/master/example/gluon/embedding_learning/model.py#L61

**Comment by TitleZ99** [comment]
Thank you very much for the timely response, I will learn from the response you have given.
